securesystemslib/keys.py

```diff
@@ -803,14 +801,15 @@ def verify_signature(key_dict, signature, data):
   formats.ANYKEY_SCHEMA.check_match(key_dict)
 
   # Does 'signature' have the correct format?
-  formats.SIGNATURE_SCHEMA.check_match(signature)
+  if not isinstance(signature, signer.Signature):
+    raise exceptions.FormatError("Wanted a Signature object.")
```

Comment:
Nit, feel free to ignore if you decide:
A comment I once got a while ago: `Don't add dots or exclamation marks at the end of an exception message`.